functests/dpdk/dpdk.go

```diff
@@ -414,7 +414,7 @@ func findOrOverrideSriovPolicyAndNetwork() {
 
 		Expect(len(nn)).To(BeNumerically(">", 0))
 
-		sriovDevice, err := sriovInfos.FindOneSriovDevice(nn[0])
+		sriovDevice, err := FindDpdkSriovDevice(sriovInfos, nn[0])
```

Comment:
Do we need the same change in discovery mode? See https://github.com/openshift-kni/cnf-features-deploy/blob/4c169cd5bdce2a098c0923934917395ca4fe2985/functests/utils/discovery/discovery.go#L69

Reply:
No I want to leave the user the ability to test on a different interfaces that are not the default gateway interface.

In general in discovery mode the user should know why he creates the policy on a specific interface.

Reply:
Yep, but even in discovery mode there will be interfaces belonging to the pf where the default gw is, and others belonging to other pfs. Would it make sense to favour the first one?

Reply:
Let me fix the discovery in a separate PR, I need to cherry-pick this to release 4.5

Reply:
Agree